examples/apps/colorapp/src/colorteller/deploy.sh

```diff
@@ -14,12 +13,25 @@ if [ -z $AWS_DEFAULT_REGION ]; then
 fi
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
-COLOR_TELLER_IMAGE=${COLOR_TELLER_IMAGE:-"${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/colorteller"}
+ECR_URL="${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com"
+COLOR_TELLER_IMAGE=${COLOR_TELLER_IMAGE:-"${ECR_URL}/colorteller"}
 GO_PROXY=${GO_PROXY:-"https://proxy.golang.org"}
+AWS_CLI_VERSION=$(aws --version 2>&1 | cut -d/ -f2 | cut -d. -f1)
+
+ecr_login() {
+        if [ $AWS_CLI_VERSION -eq 1 ]; then
+            $(aws ecr get-login --no-include-email)
+        elif [ $AWS_CLI_VERSION -eq 2 ]; then
+            aws ecr get-login-password | docker login --username AWS --password-stdin ${ECR_URL}
```

Comment:
Would it make more sense to use `get-login-password` if `$AWS_CLI_VERSION was > 1`, else `get-login`? I don't see this as guaranteed to change again in AWS CLI v3.

Reply:
Yeah, I don't see it changing again any time soon. I just used the few examples that were already in place and assumed it was a way to safeguard against it changing again in a future v3. Also it gives us a safety trip if for some reason v2+n does come along and we've not updated these passages. That was my reasoning to keep it the same as the precedent, but if there's a benefit I'd be happy to change it.

Reply:
My particular concern is that this is guaranteed to result in "broken" demos when v3 comes out, as opposed to optimistically assuming `get-login-password` will work the same in v3. I think we should favor a durable change here so that we aren't forced to revisit it when a new version is released.

Reply:
Works for me.